src/app/ReadHandler.cpp

```diff
@@ -61,6 +62,55 @@ ReadHandler::ReadHandler(ManagementCallback & apCallback, Messaging::ExchangeCon
     mSessionHandle.Grab(mExchangeCtx->GetSessionHandle());
 }
 
+ReadHandler::ReadHandler(ManagementCallback & apCallback, SubscriptionResumptionStorage::SubscriptionInfo & subscriptionInfo) :
+    mExchangeCtx(*this), mManagementCallback(apCallback), mOnConnectedCallback(HandleDeviceConnected, this),
+    mOnConnectionFailureCallback(HandleDeviceConnectionFailure, this)
+{
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    mInteractionType         = InteractionType::Subscribe;
+    mSubscriptionId          = subscriptionInfo.mSubscriptionId;
+    mMinIntervalFloorSeconds = subscriptionInfo.mMinInterval;
+    mMaxInterval             = subscriptionInfo.mMaxInterval;
+    SetStateFlag(ReadHandlerFlags::FabricFiltered, subscriptionInfo.mFabricFiltered);
+
+    CHIP_ERROR err;
+    for (size_t i = 0; i < subscriptionInfo.mAttributePaths.AllocatedCount(); i++)
+    {
+        AttributePathParams attributePathParams = subscriptionInfo.mAttributePaths[i].GetParams();
+        err = InteractionModelEngine::GetInstance()->PushFrontAttributePathList(mpAttributePathList, attributePathParams);
+        if (err != CHIP_NO_ERROR)
+        {
+            Close();
+            return;
+        }
+    }
+    for (size_t i = 0; i < subscriptionInfo.mEventPaths.AllocatedCount(); i++)
+    {
+        EventPathParams eventPathParams = subscriptionInfo.mEventPaths[i].GetParams();
+        err = InteractionModelEngine::GetInstance()->PushFrontEventPathParamsList(mpEventPathList, eventPathParams);
+        if (err != CHIP_NO_ERROR)
+        {
+            Close();
+            return;
+        }
+    }
```

Comment:
Why not call default constructor if `CHIP_CONFIG_PERSIST_SUBSCRIPTIONS` not supported?

Reply:
This constructor has different arguments and is only called when `CHIP_CONFIG_PERSIST_SUBSCRIPTIONS` is supported.

This `#if` is to eliminate code when not supported. I suppose I could move this to around the entire constructor and its declaration.

Reply:
If it's just for the subs case, could we just have a different method than the constructor? It doesn't need to be the constructor and that would help with the other error handling comments given.